src/github.com/matrix-org/dendrite/clientapi/routing/room_tagging.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/accounts"
+	"github.com/matrix-org/dendrite/clientapi/httputil"
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/dendrite/common"
+	"github.com/matrix-org/gomatrixserverlib"
+	"github.com/matrix-org/util"
+)
+
+// NewMTag creates and returns a new MTag type variable
+func NewMTag() common.MTag {
+	return common.MTag{
+		Tags: make(map[string]common.TagProperties),
+	}
+}
+
+// GetTag implements GET /_matrix/client/r0/user/{userID}/rooms/{roomID}/tags
+func GetTag(req *http.Request, accountDB *accounts.Database, userID string, roomID string) util.JSONResponse {
+
+	if req.Method != http.MethodGet {
+		return util.JSONResponse{
+			Code: http.StatusMethodNotAllowed,
+			JSON: jsonerror.NotFound("Bad method"),
+		}
+	}
+
+	mtag := NewMTag()
+
+	localpart, _, err := gomatrixserverlib.SplitID('@', userID)
+	if err != nil {
+		return httputil.LogThenError(req, err)
+	}
+
+	data, err := accountDB.GetAccountDataByType(
+		req.Context(), localpart, roomID, "m.tag",
+	)
+
+	if err != nil {
+		return httputil.LogThenError(req, err)
+	}
+
+	dataByte, _ := json.Marshal(data)
+	err = json.Unmarshal(dataByte, &mtag)
```

Comment:
This seems like a sorcery. Maybe there a straightforward way to copy data into mtag?

Reply:
Um the problem was "data" from the DB was in `gomatrixserverlib.Client` form and I had to Marshal it. Is there another way?

Reply:
`gomatrixserverlib.Client.Content` contains the mtag which is mashaled JSON.
I think you can return that in the `util.JSONResponse` directly.

Reply:
I asked @babolivier what I can do about it since I wasnt clear. So I proceeded with it, Ill look into this now :)

Reply:
Also, its `type []gomatrixserverlib.ClientEvent` my bad.

Reply:
@anoadragon453 what about this?

Reply:
One option is to just cast from one type to another, so `mtag = MTag(data)` though not 100% sure if that'll work.

Reply:
Yeah this doesn't work. I'll just stick to what I've already done